src/session/session_api.c

```diff
@@ -1577,6 +1577,7 @@ int
 __wt_session_range_truncate(
   WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *start, WT_CURSOR *stop)
 {
```

Comment:
Can we ensure we pass the file cursor to this function? We should do the conversion in the upstream.

Reply:
rts_history.c
```
  if (!dryrun)
        WT_ERR(truncate_session->truncate(
          truncate_session, NULL, hs_cursor_start, hs_cursor_stop, NULL));
```

Reply:
I'm not sure that we can assert that only {{WT_CURSOR_BTREE}} is passed to this function given that this is called pretty much directly from `WT_SESSION::truncate`, but if we are looking for a quick 8.0 fix, we could just change these lines in `rts_history.c` and add a FIXME to do a proper check for whether bounds are supported (as I explained in another comment).